Geometry_Engine/Modify/RoundCoordinates.cs

```diff
@@ -0,0 +1,324 @@
+        [Input("polycurve", "The BHoM Geometry PolyCurve to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("curve", "The modified BHoM Geometry PolyCurve.")]
+        public static PolyCurve RoundCoordinates(this PolyCurve polycurve, int decimalPlaces = 6)
+        {
+            return new PolyCurve()
+            {
+                Curves = polycurve.Curves.Select(x => x.IRoundCoordinates(decimalPlaces)).ToList()
+            };
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry NurbsCurve's control points to be rounded to the number of provided decimal places.")]
+        [Input("nurbscurve", "The BHoM Geometry NurbsCurve to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("curve", "The modified BHoM Geometry NurbsCurve.")]
+        public static NurbsCurve RoundCoordinates(this NurbsCurve nurbscurve, int decimalPlaces = 6)
+        {
+            return new NurbsCurve()
+            {
+                ControlPoints = nurbscurve.ControlPoints.Select(x => x.RoundCoordinates(decimalPlaces)).ToList(),
+                Knots = nurbscurve.Knots.ToList(),
+                Weights = nurbscurve.Weights.ToList()
+            };
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry PlanarSurface's defining curves to be rounded to the number of provided decimal places.")]
+        [Input("nurbscurve", "The BHoM Geometry PlanarSurface to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("surface", "The modified BHoM Geometry PlanarSurface.")]
+        public static PlanarSurface RoundCoordinates(this PlanarSurface planarSurface, int decimalPlaces = 6)
+        {
+            Vector normal = planarSurface.Normal().Normalise();
+
+            if (Math.Abs(Math.Abs(normal.X) - 1) < Tolerance.Angle ||
+                Math.Abs(Math.Abs(normal.Y) - 1) < Tolerance.Angle ||
+                Math.Abs(Math.Abs(normal.Z) - 1) < Tolerance.Angle)
+            {
+                Plane plane = new Plane() { Origin = planarSurface.ExternalBoundary.IStartPoint().RoundCoordinates(decimalPlaces), Normal = normal.RoundCoordinates(0) };
+                ICurve externalBoundary = planarSurface.ExternalBoundary.IProject(plane).IRoundCoordinates(decimalPlaces);
+                List<ICurve> internalBoundaries = planarSurface.InternalBoundaries.Select(x => x.IProject(plane).IRoundCoordinates(decimalPlaces)).ToList();
+
+                return Create.PlanarSurface(externalBoundary, internalBoundaries);
+            }
+
+            Reflection.Compute.RecordWarning("Rounding the coordinates of a non-Axis-aligned planar surface while maintaining planarity needs to be done manually. No action has been taken.");
```

Comment:
Could slightly rephrase that to e.g.:

`Reflection.Compute.RecordWarning("Rounding the coordinates of a planar surface that is not aligned with global coordinate system cannot be achieved without risk of losing planarity. No action has been taken.");`

Reply:
Is it worth clarifying `the global coordinate system` or `a global coordinate system`?